evm/src/executor/inspector/cheatcodes/util.rs

```diff
@@ -120,6 +124,24 @@ fn sign(private_key: U256, digest: H256, chain_id: U256) -> Result {
     Ok((sig.v, r_bytes, s_bytes).encode().into())
 }
 
+fn create_wallet(private_key: U256, label: Option<String>, state: &mut Cheatcodes) -> Result {
+    let key = parse_private_key(private_key)?;
+    let addr = utils::secret_key_to_address(&key);
+
+    let pub_key = key.verifying_key().as_affine().to_encoded_point(false);
+    let pub_key_x = pub_key.x().ok_or("No x coordinate was found")?;
+    let pub_key_y = pub_key.y().ok_or("No y coordinate was found")?;
+
+    let pub_key_x = U256::from(pub_key_x.as_slice());
+    let pub_key_y = U256::from(pub_key_y.as_slice());
+
+    if let Some(label) = label {
+        state.labels.insert(addr, label);
+    }
+
+    Ok((addr, pub_key_x, pub_key_y, private_key).encode().into())
```

Comment:
Ideally, `private_key` or `addr` should be reconstructed with `pub_key_x` and `pub_key_y` to ensure that the coordinates are correct, but I couldn't find a great way do that yet. I may make another commit adding that and reference that commit here, but I'm currently looking for suggestions 👍

Reply:
I think it's ok right now to construct the `addr` from the private key—is the usual way you see it implemented in most libraries, even if it should be possible to construct it from the pubkeys—constructing from the privkey is just one more step.